libnmstate/nm/device.py

```diff
@@ -20,6 +20,96 @@
 from . import nmclient
 
 
+class ActivationError(Exception):
+    pass
+
+
+class AlternativeACState(object):
+    UNKNOWN = 0
+    ACTIVE = 1
+    ACTIVATING = 2
+    FAIL = 3
+
+
+class ActiveConnection(object):
+
+    def __init__(self, active_connection):
+        self._act_con = active_connection
+        self._state = None
+        self._state_reason = nmclient.NM.ActiveConnectionStateReason.UNKNOWN
+        self._alternative_state = AlternativeACState.UNKNOWN
+        nmdevs = active_connection.get_devices()
+        self._nmdev = nmdevs[0] if nmdevs else None
+
+        self.handlers = set()
+
+    def refresh_state(self):
+        self._state = self._act_con.get_state()
+        self._state_reason = self._act_con.get_state_reason()
+        self._alternative_state = (
+            nmclient.NM.ActiveConnectionStateReason.UNKNOWN)
+
+        nm_acs = nmclient.NM.ActiveConnectionState
+        if self._state == nm_acs.DEACTIVATED:
+            unable_to_activate = (
+                    not self._nmdev or
+                    (
+                            self._state_reason is not None and
+                            self._state_reason != nm_acs.DEVICE_DISCONNECTED
+                    ) or
+                    self._nmdev.get_active_connection() is not self._act_con
+            )
+            if unable_to_activate:
+                self._alternative_state = AlternativeACState.FAIL
+            # Use the device-state as an alternative to determine if active.
+            elif (self._nmdev <= nmclient.NM.DeviceState.DISCONNECTED or
+                    self._nmdev > nmclient.NM.DeviceState.DEACTIVATING):
+                self._alternative_state = AlternativeACState.FAIL
+
+    @property
+    def is_active(self):
```

Comment:
The `is_` prefix does not make sense for properties IMHO. Being a property implies that it is somthing that the class is.

Reply:
My suggestion is to rename it to `active`.